File: iOS/TUILiveKit/Source/View/LiveIdentityGenerator.swift
//
//  LiveIdentityGenerator.swift
//  TUILiveKit
//
//  Created by WesleyLei on 2024/6/6.
//

import Foundation

public final class LiveIdentityGenerator {
    public enum RoomType: CaseIterable {
        case live
        case voice
        var prefix: String {
            switch self {
            case .live:
                return "live_"
            case .voice:
                return "voice_"
            }
        }
    }

    public static let shared = LiveIdentityGenerator()
    private init() {
    }

    public func generateId(_ id: String, _ type: RoomType) -> String {
        return type.prefix + id
    }

    public func getIDType(_ id: String) -> RoomType? {
        for type in RoomType.allCases {
            if id.hasPrefix(type.prefix) {
                return type
            }
        }
        return nil
    }
}

File: Android/tuilivekit/src/main/java/com/trtc/uikit/livekit/common/LiveIdentityGenerator.kt
package com.trtc.uikit.livekit.common

object LiveIdentityGenerator {
    fun generateId(id: String, type: RoomType) = type.prefix + id

    enum class RoomType(val prefix: String) {
        LIVE("live_"),
        VOICE("voice_")
    }
}